java-extensions/iceberg-metadata-reader/src/main/java/com/starrocks/connector/iceberg/IcebergRefsTableScanner.java

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.iceberg;
+
+import com.starrocks.jni.connector.ColumnType;
+import com.starrocks.jni.connector.ColumnValue;
+import com.starrocks.jni.connector.ConnectorScanner;
+import com.starrocks.utils.loader.ThreadContextClassLoader;
+import org.apache.iceberg.SnapshotRef;
+import org.apache.iceberg.Table;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import static org.apache.iceberg.util.SerializationUtil.deserializeFromBase64;
+
+public class IcebergRefsTableScanner extends ConnectorScanner {
+    private static final Logger LOG = LogManager.getLogger(IcebergMetadataScanner.class);
+
+    private final String serializedTable;
+    private final String[] requiredFields;
+    private final String[] metadataColumnNames;
+    private final String[] metadataColumnTypes;
+    private ColumnType[] requiredTypes;
+    private final int fetchSize;
+    private final ClassLoader classLoader;
+    private Table table;
+    private Iterator<Map.Entry<String, SnapshotRef>> reader;
+
+    public IcebergRefsTableScanner(int fetchSize, Map<String, String> params) {
```

Comment:
read refs from iceberg could be time-consuming? why not just read in FE

Reply:
Calling fe is easy to cause a single point bottleneck, and it is also convenient to keep the architecture clean